Pipfile

```diff
@@ -11,15 +11,15 @@ gunicorn = "~=22.0"
 social-auth-app-django = "~=5.4"
 requests = "~=2.31"
 psutil = "~=5.9"
-celery = "~=5.3.5"  # Celery v5.4 may stop supporting Python 3.8
+celery = "~=5.4"
 django-celery-results = "~=2.5"
 django-markdownify = "~=0.9"  # includes the markdown package
 psycopg = "~=3.1"
 virtualenv = "~=20.26"
 redis = "~=5.0"
 channels-redis = "~=4.2"
 django-extensions = "~=3.2"
-ipython = "~=8.12.3"  # IPython follows NEP 29, so v8.13 does not support Python 3.8
+ipython = "~=8.24"
```

Comment:
I think you'll want to define the patch version here (and also for `celery`).

This is because when you define `~=` constraints with pipenv, it'll constrain the version to the second-last defined number. So, `~=8.12.3` will be `8.12.*`, and `~=8.24` will be `8.*`. Since `ipython` and `celery` change Python version compatibility with minor releases (unlike standard semver), defining an unchanging minor version will keep the packages compatible in the future.

tl;dr - use `ipython = "~=8.24.0"` and `celery = "~=5.4.0"`